gk/main.c

```diff
@@ -952,15 +974,13 @@ process_pkts_front(uint8_t port_front, uint8_t port_back,
 						packet.flow.f.v6.dst);
 				}
 
-				if (eth_cache == NULL || eth_cache->stale) {
+				RTE_VERIFY(eth_cache != NULL);
+				if (pkt_copy_cached_eth_header(
+						pkt, eth_cache)) {
 					drop_packet(pkt);
 					continue;
 				}
 
-				eth_hdr = rte_pktmbuf_mtod(
-					pkt, struct ether_hdr *);
-				rte_memcpy(eth_hdr,
-					&eth_cache->eth_hdr, sizeof(*eth_hdr));
 				tx_bufs[num_tx++] = pkt;
```

Comment:
I just realized that we are not decrementing the TTL/hop_count field and, in the case of IPv4, recalculating the checksum while forwarding packets between interfaces. In order to avoid complicating this patch, could you write another patch to do this? If you find that it'll take time, you can put this other patch in another pull request.

Reply:
Good catch! Sure, will do that.